DonoGear/app/src/main/java/com/example/donogear/actionpages/ProductDetails.java

```diff
@@ -0,0 +1,314 @@
+            params.setMargins(5, 5, 5, 5);
+            ImageView imageView = new ImageView(context);
+            imageView.setLayoutParams(params);
+            Bitmap bitmap = BitmapFactory.decodeFile(image.toString());
+            imageView.setScaleType(ImageView.ScaleType.FIT_XY);
+            imageView.setImageBitmap(bitmap);
+            layout.addView(imageView);
+        }
+    }
+
+    /**
+     * Queries the database to get videos for the selected item
+     * @return - list of URL's of all videos for items
+     */
+    private List<String> getItemVideos() {
+        itemVideosQuery = ParseQuery.getQuery(COLLECTIBLE_VIDEOS);
+        itemVideosQuery.whereEqualTo("collectibleId", itemId);
+        System.out.println(itemId);
+        final List<String> allVideos = new ArrayList<>();
+        itemVideosQuery.getFirstInBackground((object, e) -> {
+            if (e == null) {
+                for (int i = 1; i < 4; i++) {
+                    if (object.has("video" + i)) {
+                        if (object.getString("video" + i) != null) {
+                            String videoUrl = object.getString("video" + i);
+                            System.out.println(videoUrl);
+                            allVideos.add(videoUrl);
+                        }
+                    }
+                }
+            }
+            hasVideos = true;
+        });
+        return allVideos;
+    }
+
+    /**
+     * Get the proceed details for a single item
+     */
+    private void getItemProceedsDetails() {
+        proceedsDetails = new ItemProceedsDetails();
+        itemProceedsQuery = ParseQuery.getQuery(PROCEEDS);
+        itemProceedsQuery.whereEqualTo("collectibleId", itemId);
+        itemProceedsQuery.getFirstInBackground((object, e) -> {
+            if (e == null) {
+                final String proceedTitle = object.getString("proceedTitle");
+                System.out.println("Title " + proceedTitle);
+                final String proceedDescription = object.getString("proceedDescription");
+                System.out.println("Desc " + proceedDescription);
+                final List<File> proceedsImages = new ArrayList<>();
+                final List<String> proceedVideo = new ArrayList<>();
+                for (int i = 1; i < 3; i++) {
+                    if (object.getParseFile("proceedImage" + i) != null) {
+                        try {
+                            if (object.getParseFile("proceedImage" + i).getFile() != null) {
+                                File image = object.getParseFile("proceedImage" + i).getFile();
+                                System.out.println("Added image " + i);
+                                proceedsImages.add(image);
+                            }
+                        } catch (ParseException ex) {
+                            ex.printStackTrace();
+                        }
+                    }
+                }
+                for (int i = 1; i < 2; i++) {
```

Comment:
I think a for loop is overkill here if there's only going to be 2 loops. If the same functionality is needed twice, just create a method for it and invoke it twice here